ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -984,14 +990,14 @@ boolean isComplete() {
 
     // refers to the blob type.
     private BlobType blobType;
-    private List<StoreKey> keys;
+    private List<CompositeBlobInfo.StoreKeyAndSizeAndOffset> keysAndSizesAndOffsets;
     private BlobProperties serverBlobProperties;
 
     /**
      * Construct a FirstGetChunk and initialize it with the {@link BlobId} of the overall operation.
      */
     FirstGetChunk() {
-      super(-1, blobId);
+      super(-1, new CompositeBlobInfo.StoreKeyAndSizeAndOffset(blobId, 0L, -1L));
```

Comment:
It seems the `chunkSize` is initialized to `-1L` here and I don't see it gets populated again because you removed several codes regarding setting `chunkSize`.  Correct me if I am wrong.

Reply:
That's right, but that's ok since chunkSize is only used when calculating the end of data chunk for a byte range operation if that data chunk is not the last data chunk.  For simple data blobs, the FirstChunk will always be the last data chunk so it's unused, and for metadata blobs the size of the FirstChunk is irrelevant.